client/task-list/tasks/payments/RemotePayments/components/RecommendedPaymentGatewayList/RecommendedPaymentGatewayList.js

```diff
@@ -1,122 +1,51 @@
-			const {
-				image,
-				content,
-				key,
-				plugins = [],
-				title,
-				is_visible: isVisible,
-				loading,
-			} = method;
-
-			if ( ! isVisible ) {
-				return null;
+} ) => {
+	const recommendedMethod = useMemo( () => {
+		for ( const key in RECOMMENDED_GATEWAY_KEYS ) {
+			const gateway = recommendedPaymentGateways.get( key );
+			if ( gateway && gateway.visible ) {
+				return gateway;
 			}
-
-			const installedPaymentGateway = installedPaymentGateways[ key ];
-			const {
-				enabled: isEnabled = false,
-				needs_setup: needsSetup = false,
-				required_settings_keys: requiredSettingsKeys = [],
-				settings_url: manageUrl,
-			} = installedPaymentGateway || {};
-
-			const isConfigured = ! needsSetup;
-			const hasSetup = Boolean(
-				plugins.length || requiredSettingsKeys.length
-			);
-			const isRecommended = key === recommendedMethod && ! isConfigured;
-			const showRecommendedRibbon = isRecommended;
-
-			const classes = classnames(
-				'woocommerce-task-payment',
-				'woocommerce-task-card',
-				! isConfigured && 'woocommerce-task-payment-not-configured',
-				'woocommerce-task-payment-' + key
-			);
-
-			return (
-				<Fragment key={ key }>
-					{ index !== 0 && <CardDivider /> }
-					<CardBody
-						style={ { paddingLeft: 0, marginBottom: 0 } }
-						className={ classes }
-					>
-						<CardMedia isBorderless>
-							<img src={ image } alt={ title } />
-						</CardMedia>
-						<div className="woocommerce-task-payment__description">
-							{ showRecommendedRibbon && <RecommendedRibbon /> }
-							<Text
-								as="h3"
-								className="woocommerce-task-payment__title"
-							>
-								{ title }
-								{ isEnabled && ! isConfigured && (
-									<SetupRequired />
-								) }
-							</Text>
-							<div className="woocommerce-task-payment__content">
-								{ content }
-							</div>
-						</div>
-						<div className="woocommerce-task-payment__footer">
-							<PaymentAction
-								manageUrl={ manageUrl }
-								methodKey={ key }
-								hasSetup={ hasSetup }
-								isConfigured={ isConfigured }
-								isEnabled={ isEnabled }
-								isRecommended={ isRecommended }
-								isLoading={ loading }
-								markConfigured={ markConfigured }
-								onSetup={ () =>
-									recordEvent( 'tasklist_payment_setup', {
-										options: recommendedPaymentGateways.map(
-											( option ) => option.key
-										),
-										selected: key,
-									} )
-								}
-								onSetupCallback={ method.onClick }
-							/>
-						</div>
-					</CardBody>
-				</Fragment>
-			);
-		} ) }
-	</Card>
-);
+		}
+		return null;
+	}, [ recommendedPaymentGateways ] );
+
+	return (
+		<Card>
+			<CardHeader as="h2">{ heading }</CardHeader>
+			{ Array.from( recommendedPaymentGateways.values() ).map(
+				( paymentGateway, index ) => {
+					const { key } = paymentGateway;
+					return (
+						<RecommendedPaymentGatewayListItem
+							hasDivider={ index !== 0 }
```

Comment:
I don't think this is being set properly, since `recommendedPaymentGateways` includes those that won't be visible.

Reply:
Good call!  Updated to a simple CSS solution in https://github.com/woocommerce/woocommerce-admin/pull/7073/commits/790fd74b87787132cd325b36548d7cd3e73b1fa8